src/backend/access/transam/transam.c

```diff
@@ -170,6 +171,41 @@ TransactionIdDidCommit(TransactionId transactionId)
 	return false;
 }
 
+/*
+ * RemoteTransactionIdDidCommit
+ *		True iff transaction associated with the remote identifier did commit.
+ *
+ * Note:
+ *		Assumes transaction identifier is valid and exists in csn_log.
+ *
+ * Remotexact
+ */
+bool							/* true if given transaction committed */
+RemoteTransactionIdDidCommit(int region, TransactionId transactionId)
+{
+	XidCSN csn;
+
+	// TODO (ctring): cache the fetched id like in TransactionLogFetch?
+	/*
+	 * Also, check to see if the transaction ID is a permanent one.
+	 */
+	if (!TransactionIdIsNormal(transactionId))
+	{
+		if (TransactionIdEquals(transactionId, BootstrapTransactionId))
+			return true;
+		if (TransactionIdEquals(transactionId, FrozenTransactionId))
+			return true;
+		return false;
+	}
+
+	/*
+	 * Get the csn corresponding to the transaction id.
+	 */
+	csn = CSNLogGetCSNByXid(region, transactionId);
```

Comment:
Here: shouldn't we be considering if the csn <= GetRegionLsn(region)? Otherwise even if this transaction has committed at a later point, we still need to return false in order to execute at a consistent snapshot.

Reply:
Nice catch! I forgot that the CSN page can be in the future